src/SonarScanner.MSBuild.TFS/IBuildVNextCoverageSearchFallback.cs

```diff
@@ -74,16 +75,28 @@ public IEnumerable<string> FindCoverageFiles()
                 Logger.LogInfo($"No coverage files found in the agent temp directory.");
                 return Enumerable.Empty<string>();
             }
-            else
-            {
-                LogDebugFileList("All matching files:", files);
 
-                // The same file might appear in multiple paths.
-                // We're assuming the files are identical so it doesn't matter which one we pick.
-                files = files.Distinct(new FileNameComparer()).ToArray();
-                LogDebugFileList("Unique coverage files:", files);
-                return files;
-            }
+            LogDebugFileList("All matching files:", files);
+
+            var fileWithContentHashes = files.Select(fullFilePath =>
+            {
+                using (var fileStream = new FileStream(fullFilePath, FileMode.Open))
+                using (var bufferedStream = new BufferedStream(fileStream))
+                using (var sha = new SHA256Managed())
+                {
+                    var contentHash = sha.ComputeHash(bufferedStream).ToString();
```

Comment:
Why don't you keep this one as `byte[]` and then do a byte-by-byte comparison in `FileHashComparer`? It would be more efficient.